qiskit/opflow/gradients/gradient.py

```diff
@@ -152,7 +160,7 @@ def is_coeff_c(coeff, c):
         if isinstance(operator, ComposedOp):
 
             # Gradient of an expectation value
-            if not is_coeff_c(operator._coeff, 1.0):
+            if not is_coeff_c_abs(operator._coeff, 1.0):
```

Comment:
Why do we only care about the absolute value being one here? Couldn't we rather check if the coeff is 1 or i instead of the absolute value? That would also not require the new `is_coeff_c_abs` function 🙂

Reply:
We could but it's also fine if there is a sign. Extracting the sign and adding it again may be uncleaner than just checking for the absolute value. What do you think?